### ISSUE TITLE
Search through tags

### ISSUE THREAD

**Issue by pierrocknroll** [cviebrock/eloquent-taggable#69]
Hello,

I'm wondering how to implement with efficienty a functionality to search through tags for an autocomplete field.
For exemple localhost/tags?query=lorem searches all tags who starts by lorem

I haven't seen such a thing in your readme.

Thanks !

**Comment by cviebrock** [comment]
Wouldn’t something like `Tags::whereLike(‘name’, ‘lorem%’)->get()` work?

**Comment by pierrocknroll** [comment]
Yeah, of course !
I rather thought about implemeting it with Scout.

**Comment by pierrocknroll** [comment]
I try to do something like : 
Model::allTags()->where('name', 'LIKE', 'lorem%').

But your methods always return a Collection (with Tag::fromQuery($sql)). Is it possible to have a method who returns a QueryBuilder ? Or another way to add SQL to your queries.

Thanks

**Comment by cviebrock** [comment]
Yeah, it's not ideal, I agree.  This package needs a bit of attention and probably a bit of a rewrite.
